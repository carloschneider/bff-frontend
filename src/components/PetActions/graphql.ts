import { gql } from '@apollo/client'

export type CheckType = {
  id: string
}

export type CheckInDataType = {
  checkIn: CheckType[]
}

export type CheckVariablesType = {
  petId: string
}

export const MUTATION_CHECKIN = gql`
  mutation CheckIn($petId: String!) {
    checkIn(petId: $petId) {
      id
    }
  }
`

export type CheckOutDataType = {
  checkIn: CheckType[]
}

export const MUTATION_CHECKOUT = gql`
  mutation CheckOut($petId: String!) {
    checkOut(petId: $petId) {
      id
    }
  }
`
