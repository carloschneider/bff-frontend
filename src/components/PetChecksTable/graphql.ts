import { gql } from '@apollo/client'

import { PaginationType } from 'pages/Pets'

export type StaffType = {
  id: string
  firstName: string
  lastName: string
}

export type UserType = {
  id: string
  companyName: string
}

export type ResponsibleType = {
  id: string
  type: 'arrive' | 'leave'
  staff: StaffType | null
  user: UserType | null
}

export type CheckType = {
  id: string
  arrive: string
  leave: string | null
  count?: number | null
  responsibles: ResponsibleType[]
}

export type ChecksDataType = {
  getAllChecksByPetId: CheckType[] | undefined
}

type ChecksWhereType = {
  id?: string
}

export type ChecksVariablesType = {
  where?: ChecksWhereType | null
} & PaginationType

export const GET_CHECKS_BY_PET_ID = gql`
  query GetAllChecksByPetId(
    $limit: Int
    $offset: Int
    $field: String
    $order: String
    $where: CheckWhereGetAllByPetIdInput!
  ) {
    getAllChecksByPetId(
      limit: $limit
      offset: $offset
      field: $field
      order: $order
      where: $where
    ) {
      id
      arrive
      leave
      count
      responsibles {
        id
        type
        staff {
          id
          firstName
          lastName
        }
        user {
          id
          companyName
        }
      }
    }
  }
`
