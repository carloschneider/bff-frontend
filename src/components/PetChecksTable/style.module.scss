.wrap {
  width: 100%;
}

.buttons {
  align-items: center;
  display: flex;
  justify-content: end;
}
