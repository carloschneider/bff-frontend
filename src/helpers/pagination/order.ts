export enum OrderEnum {
  DESC = 'DESC',
  ASC = 'ASC'
}

export const convertOrder = (order: string): OrderEnum => {
  const desc = OrderEnum.DESC
  const asc = OrderEnum.ASC

  if (order.includes(desc.toLowerCase())) {
    return desc
  }

  return asc
}
