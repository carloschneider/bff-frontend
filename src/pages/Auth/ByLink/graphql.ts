import { gql } from '@apollo/client'

export type AuthByKeyType = {
  token: string
  role: string
}

export type AuthByKeyDataType = {
  authByKey: AuthByKeyType
}

export type AuthByKeyInputType = {
  key: string
}

export type AuthByKeyVariablesInputType = {
  input: AuthByKeyInputType
}

export const MUTATION_AUTH_BY_KEY = gql`
  mutation AuthByKeyInput($input: AuthByKeyInput!) {
    authByKey(input: $input) {
      token
      role
    }
  }
`
