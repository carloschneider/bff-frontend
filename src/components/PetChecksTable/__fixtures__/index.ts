import { CheckType } from '../graphql'

export const getAllChecksByPetId: CheckType[] = [
  {
    id: 'a6a3300f-fcf8-40cc-89f2-1fa85d4c4ba2',
    arrive: '2023-01-25T18:26:43.000Z',
    leave: '2023-01-25T18:27:36.000Z',
    count: 6,
    responsibles: [
      {
        id: '9a6f4710-fd3e-49ac-a0f8-d38498e7b45b',
        type: 'arrive',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      },
      {
        id: '62ad42eb-699c-4b8a-a0a3-c0f25189acaa',
        type: 'leave',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      }
    ]
  },
  {
    id: '4b63fd9c-d49b-4c5d-8c38-1fe72a6e2f88',
    arrive: '2023-01-07T17:31:36.000Z',
    leave: '2023-01-07T17:31:38.000Z',
    count: null,
    responsibles: [
      {
        id: 'cdee01c6-044a-47a3-a0f5-8b94fa59ee3d',
        type: 'leave',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      },
      {
        id: 'b9d79e38-bc04-4674-b098-ee00f858e57d',
        type: 'arrive',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      }
    ]
  },
  {
    id: 'ade1d25c-0802-4aca-8524-0f493a609dc2',
    arrive: '2023-01-07T17:29:39.000Z',
    leave: '2023-01-07T17:29:41.000Z',
    count: null,
    responsibles: [
      {
        id: 'b82f9601-1660-4f0a-a19c-f10237868916',
        type: 'leave',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      },
      {
        id: '6eea4caa-445d-42b8-b196-dc9fdf25f1f8',
        type: 'arrive',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      }
    ]
  },
  {
    id: '8846501b-c64d-4708-85e1-ba028e02430c',
    arrive: '2023-01-07T03:35:02.000Z',
    leave: '2023-01-07T03:35:04.000Z',
    count: null,
    responsibles: [
      {
        id: 'be15638d-f468-4f25-837d-a384c37c2464',
        type: 'arrive',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      },
      {
        id: '18be9589-8c4e-4dd6-9fa2-ef5a0a4f2876',
        type: 'leave',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      }
    ]
  }
]

export const getAllChecksByPetIdPageTwo: CheckType[] = [
  {
    id: 'ade1d25c-0802-4aca-8524-0f493a609dc2',
    arrive: '2023-01-07T17:29:39.000Z',
    leave: '2023-01-07T17:29:41.000Z',
    count: 8,
    responsibles: [
      {
        id: 'b82f9601-1660-4f0a-a19c-f10237868916',
        type: 'leave',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      },
      {
        id: '6eea4caa-445d-42b8-b196-dc9fdf25f1f8',
        type: 'arrive',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      }
    ]
  },
  {
    id: '8846501b-c64d-4708-85e1-ba028e02430c',
    arrive: '2023-01-07T03:35:02.000Z',
    leave: '2023-01-07T03:35:04.000Z',
    count: 7,
    responsibles: [
      {
        id: 'be15638d-f468-4f25-837d-a384c37c2464',
        type: 'arrive',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      },
      {
        id: '18be9589-8c4e-4dd6-9fa2-ef5a0a4f2876',
        type: 'leave',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      }
    ]
  },
  {
    id: 'e0d3a00b-56d9-4667-abd6-edb62ed1a746',
    arrive: '2023-01-03T21:47:23.000Z',
    leave: '2023-01-03T21:47:32.000Z',
    count: null,
    responsibles: [
      {
        id: '0eaaf655-86c5-4ff6-946b-5fa94f1747ae',
        type: 'arrive',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      },
      {
        id: 'be9d898c-e7d9-45ca-ac19-b2824248cb98',
        type: 'leave',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      }
    ]
  },
  {
    id: 'd34065a1-53b7-40b4-a949-8c21268461bb',
    arrive: '2023-01-03T20:32:56.000Z',
    leave: '2023-01-03T20:32:58.000Z',
    count: null,
    responsibles: [
      {
        id: 'bfae6391-9bcd-403a-a54f-ea22066b8dfe',
        type: 'arrive',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      },
      {
        id: 'c2ad7b13-a5a0-4df6-9813-14300f83a880',
        type: 'leave',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      }
    ]
  }
]
