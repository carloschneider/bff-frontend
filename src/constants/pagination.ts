import { OrderEnum } from 'helpers/pagination/order'

export const PAGINATION_DEFAULT_PAGE = 1
export const PAGINATION_DEFAULT_ORDER = OrderEnum.DESC
export const PAGINATION_DEFAULT_FIELD = 'createdAt'
export const PAGINATION_DEFAULT_LIMIT = 10
export const PAGINATION_DEFAULT_OFFSET = 0
