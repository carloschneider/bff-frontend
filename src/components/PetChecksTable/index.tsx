import { useLazyQuery } from '@apollo/client'
import { Col, Pagination, Row, Space, Table, Typography } from 'antd'
import { ColumnsType, TablePaginationConfig } from 'antd/lib/table'
import { FilterValue, SorterResult } from 'antd/lib/table/interface'
import { useEffect, useState } from 'react'

import CheckResponsible from 'components/CheckResponsible'
import PetActions from 'components/PetActions'
import { convertOrder, OrderEnum } from 'helpers/pagination/order'
import { PetType } from 'pages/Pet/graphql'
import { PaginationType } from 'pages/Pets'

import {
  ChecksDataType,
  ChecksVariablesType,
  CheckType,
  GET_CHECKS_BY_PET_ID
} from './graphql'
import style from './style.module.scss'

const { Title } = Typography

type PetChecksTypeProps = {
  id: string
  name: string | undefined
}

const columnsChecks: ColumnsType<CheckType> = [
  {
    title: 'Arrive',
    dataIndex: 'arrive',
    render: (value, record: CheckType) => (
      <CheckResponsible date={value} record={record} type="arrive" />
    ),
    width: '50%'
  },
  {
    title: 'Leave',
    dataIndex: 'leave',
    render: (value, record: CheckType) => (
      <CheckResponsible date={value} record={record} type="leave" />
    ),
    width: '50%'
  }
]

const PetChecksTable = ({ id, name }: PetChecksTypeProps) => {
  const [
    getChecks,
    { data: dataChecks, loading: loadingChecks, refetch: refetchChecks }
  ] = useLazyQuery<ChecksDataType, ChecksVariablesType>(GET_CHECKS_BY_PET_ID)

  const defaultOrder = OrderEnum.DESC
  const defaultField = 'createdAt'

  const [filters, setFilters] = useState<
    ChecksVariablesType & { page: number }
  >({
    page: 1,
    order: defaultOrder,
    field: defaultField,
    limit: 4,
    offset: 0,
    where: {
      id
    }
  })

  useEffect(() => {
    getChecks({
      variables: {
        order: filters.order,
        field: filters.field,
        limit: filters.limit,
        offset: filters.offset,
        where: {
          id
        }
      },
      fetchPolicy: 'network-only',
      nextFetchPolicy: 'no-cache'
    })
  }, [filters])

  const handleChangeTable = (
    pagination: TablePaginationConfig,
    filters: Record<string, FilterValue | null>,
    sorter: SorterResult<CheckType> | SorterResult<CheckType>[]
  ) => {
    const sorterParam = sorter as SorterResult<PetType>
    const order = sorterParam.order as 'ascend' | 'descent'
    const orderConverted = convertOrder(order)
    const field = sorterParam.field as PaginationType['field']

    setFilters((prevState) => {
      return {
        ...prevState,
        order: orderConverted,
        field
      }
    })
  }

  const handleChangePagination = (pagination: number) => {
    setFilters((prevState) => {
      return {
        ...prevState,
        page: pagination,
        offset: (pagination - 1) * prevState.limit
      }
    })
  }

  return (
    <Space direction="vertical" size="middle" className={style.wrap}>
      <Row>
        <Col xs={24} sm={12} style={{ display: 'flex', alignItems: 'center' }}>
          <Title level={3} style={{ marginBottom: 0 }}>
            Checks
          </Title>
        </Col>

        <Col xs={24} sm={12} className={style.buttons}>
          <PetActions id={id} name={name} callback={refetchChecks} />
        </Col>
      </Row>

      <Table
        columns={columnsChecks}
        dataSource={dataChecks?.getAllChecksByPetId}
        rowKey={(record) => record.id}
        pagination={false}
        loading={loadingChecks}
        onChange={handleChangeTable}
        bordered
      />

      <Pagination
        current={filters.page}
        pageSize={filters.limit}
        total={
          dataChecks?.getAllChecksByPetId &&
          dataChecks?.getAllChecksByPetId.length > 0
            ? dataChecks.getAllChecksByPetId[0].count
            : 1
        }
        showSizeChanger={false}
        onChange={handleChangePagination}
        style={{ display: 'flex', justifyContent: 'end' }}
      />
    </Space>
  )
}

export default PetChecksTable
