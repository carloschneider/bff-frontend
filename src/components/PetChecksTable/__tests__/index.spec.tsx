import { MockedResponse } from '@apollo/react-testing'
import { act, fireEvent, screen, waitFor } from '@testing-library/react'

import { OrderEnum } from 'helpers/pagination/order'
import { renderWithApollo } from 'test-utils/render/renderWithApollo'

import PetChecksTable from '..'
import {
  getAllChecksByPetId,
  getAllChecksByPetIdPageTwo
} from '../__fixtures__'
import { ChecksDataType, GET_CHECKS_BY_PET_ID } from '../graphql'

describe('<PetChecksTable />', () => {
  const mockedPetId = '1'
  const mockedPetName = 'Nice Puppy'
  const apolloResponseMock: MockedResponse<ChecksDataType>[] = [
    {
      request: {
        query: GET_CHECKS_BY_PET_ID,
        variables: {
          order: OrderEnum.DESC,
          field: 'createdAt',
          limit: 4,
          offset: 0,
          where: {
            id: mockedPetId
          }
        }
      },
      result: () => {
        return {
          data: {
            getAllChecksByPetId
          }
        }
      }
    },
    {
      request: {
        query: GET_CHECKS_BY_PET_ID,
        variables: {
          order: OrderEnum.DESC,
          field: 'createdAt',
          limit: 4,
          offset: 4,
          where: {
            id: mockedPetId
          }
        }
      },
      result: () => {
        return {
          data: {
            getAllChecksByPetId: getAllChecksByPetIdPageTwo
          }
        }
      }
    }
  ]

  it('should render component', async () => {
    const { container } = renderWithApollo(
      <PetChecksTable id={mockedPetId} name={mockedPetName} />,
      apolloResponseMock
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should load inital data', async () => {
    const { container } = renderWithApollo(
      <PetChecksTable id={mockedPetId} name={mockedPetName} />,
      apolloResponseMock
    )

    await waitFor(() => {
      expect(
        container.querySelector('.ant-spin.ant-spin-spinning')
      ).toBeInTheDocument()
    })

    const mockedCheckinDate = new Date(
      getAllChecksByPetId[0].arrive
    ).toLocaleString('pt-BR', { hourCycle: 'h23' })

    const mockedCheckinMemberName = `${getAllChecksByPetId[0].responsibles[0].staff?.firstName} ${getAllChecksByPetId[0].responsibles[0].staff?.lastName}`

    const staffMember = await screen.findByText(
      `${mockedCheckinDate} (${mockedCheckinMemberName})`
    )

    expect(staffMember).toBeInTheDocument()
  })

  it('should change pagination page', async () => {
    renderWithApollo(
      <PetChecksTable id={mockedPetId} name={mockedPetName} />,
      apolloResponseMock
    )

    const mockedCheckinDate = new Date(
      getAllChecksByPetId[0].arrive
    ).toLocaleString('pt-BR', { hourCycle: 'h23' })

    const mockedCheckinMemberName = `${getAllChecksByPetId[0].responsibles[0].staff?.firstName} ${getAllChecksByPetId[0].responsibles[0].staff?.lastName}`

    const staffMember = await screen.findByText(
      `${mockedCheckinDate} (${mockedCheckinMemberName})`
    )

    expect(staffMember).toBeInTheDocument()

    const nextPageButton = screen.getByRole('button', {
      name: /right/
    })

    await act(async () => {
      await fireEvent.click(nextPageButton)
    })

    const mockedCheckinDatePageTwo = new Date(
      getAllChecksByPetIdPageTwo[0].arrive
    ).toLocaleString('pt-BR', { hourCycle: 'h23' })

    const mockedCheckinMemberNamePageTwo = `${getAllChecksByPetIdPageTwo[0].responsibles[0].staff?.firstName} ${getAllChecksByPetIdPageTwo[0].responsibles[0].staff?.lastName}`

    const staffMemberPageTwo = await screen.findByText(
      `${mockedCheckinDatePageTwo} (${mockedCheckinMemberNamePageTwo})`
    )

    expect(staffMemberPageTwo).toBeInTheDocument()
  })
})
