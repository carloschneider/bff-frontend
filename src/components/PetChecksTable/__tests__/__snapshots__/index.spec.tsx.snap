// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PetChecksTable /> should render component 1`] = `
<div
  class="ant-space css-dev-only-do-not-override-1n7nwfa ant-space-vertical wrap"
>
  <div
    class="ant-space-item"
    style="margin-bottom: 16px;"
  >
    <div
      class="ant-row css-dev-only-do-not-override-1n7nwfa"
    >
      <div
        class="ant-col ant-col-xs-24 ant-col-sm-12 css-dev-only-do-not-override-1n7nwfa"
        style="display: flex; align-items: center;"
      >
        <h3
          class="ant-typography css-dev-only-do-not-override-1n7nwfa"
          style="margin-bottom: 0px;"
        >
          Checks
        </h3>
      </div>
      <div
        class="ant-col buttons ant-col-xs-24 ant-col-sm-12 css-dev-only-do-not-override-1n7nwfa"
      >
        <div
          class="ant-space css-dev-only-do-not-override-1n7nwfa ant-space-horizontal ant-space-align-center"
        >
          <div
            class="ant-space-item"
            style="margin-right: 8px;"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-1n7nwfa ant-btn-primary ant-btn-lg"
              type="button"
            >
              <span
                class="anticon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10.854 8.146a.5.5 0 0 1 0 .708l-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 0 1 .708-.708L7.5 10.793l2.646-2.647a.5.5 0 0 1 .708 0z"
                  />
                  <path
                    d="M3.5 0a.5.5 0 0 1 .5.5V1h8V.5a.5.5 0 0 1 1 0V1h1a2 2 0 0 1 2 2v11a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2V3a2 2 0 0 1 2-2h1V.5a.5.5 0 0 1 .5-.5zM2 2a1 1 0 0 0-1 1v11a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1V3a1 1 0 0 0-1-1H2z"
                  />
                  <path
                    d="M2.5 4a.5.5 0 0 1 .5-.5h10a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5H3a.5.5 0 0 1-.5-.5V4z"
                  />
                </svg>
              </span>
              <span>
                Checkin
              </span>
            </button>
          </div>
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-1n7nwfa ant-btn-primary ant-btn-lg ant-btn-dangerous"
              type="button"
            >
              <span
                class="anticon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6.146 8.146a.5.5 0 0 1 .708 0L8 9.293l1.146-1.147a.5.5 0 1 1 .708.708L8.707 10l1.147 1.146a.5.5 0 0 1-.708.708L8 10.707l-1.146 1.147a.5.5 0 0 1-.708-.708L7.293 10 6.146 8.854a.5.5 0 0 1 0-.708z"
                  />
                  <path
                    d="M3.5 0a.5.5 0 0 1 .5.5V1h8V.5a.5.5 0 0 1 1 0V1h1a2 2 0 0 1 2 2v11a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2V3a2 2 0 0 1 2-2h1V.5a.5.5 0 0 1 .5-.5zM2 2a1 1 0 0 0-1 1v11a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1V3a1 1 0 0 0-1-1H2z"
                  />
                  <path
                    d="M2.5 4a.5.5 0 0 1 .5-.5h10a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5H3a.5.5 0 0 1-.5-.5V4z"
                  />
                </svg>
              </span>
              <span>
                Checkout
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-space-item"
  >
    <div
      class="ant-table-wrapper css-dev-only-do-not-override-1n7nwfa"
    >
      <div
        class="ant-spin-nested-loading css-dev-only-do-not-override-1n7nwfa"
      >
        <div
          class="ant-spin-container"
        >
          <div
            class="ant-table ant-table-bordered ant-table-empty"
          >
            <div
              class="ant-table-container"
            >
              <div
                class="ant-table-content"
              >
                <table
                  style="table-layout: auto;"
                >
                  <colgroup>
                    <col
                      style="width: 50%;"
                    />
                    <col
                      style="width: 50%;"
                    />
                  </colgroup>
                  <thead
                    class="ant-table-thead"
                  >
                    <tr>
                      <th
                        class="ant-table-cell"
                        scope="col"
                      >
                        Arrive
                      </th>
                      <th
                        class="ant-table-cell"
                        scope="col"
                      >
                        Leave
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="ant-table-tbody"
                  >
                    <tr
                      class="ant-table-placeholder"
                    >
                      <td
                        class="ant-table-cell"
                        colspan="2"
                      >
                        <div
                          class="css-dev-only-do-not-override-1n7nwfa ant-empty ant-empty-normal"
                        >
                          <div
                            class="ant-empty-image"
                          >
                            <svg
                              height="41"
                              viewBox="0 0 64 41"
                              width="64"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <g
                                fill="none"
                                fill-rule="evenodd"
                                transform="translate(0 1)"
                              >
                                <ellipse
                                  cx="32"
                                  cy="33"
                                  fill="#f5f5f5"
                                  rx="32"
                                  ry="7"
                                />
                                <g
                                  fill-rule="nonzero"
                                  stroke="#d9d9d9"
                                >
                                  <path
                                    d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                  />
                                  <path
                                    d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                    fill="#fafafa"
                                  />
                                </g>
                              </g>
                            </svg>
                          </div>
                          <div
                            class="ant-empty-description"
                          >
                            No data
                          </div>
                        </div>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
